pkg/controller/datavolume-controller.go

```diff
@@ -363,6 +373,28 @@ func (c *DataVolumeController) syncHandler(key string) error {
 	return nil
 }
 
+func (c *DataVolumeController) scheduleProgressUpdate(dataVolumeName, dataVolumeNamespace string) {
+	for {
+		time.Sleep(2 * time.Second)
+		dataVolume, err := c.cdiClientSet.Cdi().DataVolumes(dataVolumeNamespace).Get(dataVolumeName, metav1.GetOptions{})
+		if k8serrors.IsNotFound(err) {
+			// Data volume is no longer there, or not found.
+			glog.V(3).Info("DV is gone, cancelling update thread.")
+			return
+		} else if err != nil {
+			glog.Errorf("error retrieving data volume %+v", err)
+		}
+		if dataVolume.Status.Phase == cdiv1.Succeeded {
```

Comment:
I was thinking about our discussion around handling transient states on these, since we have this annoying case where the DV can go in and out of failed state, would it make sense to try and keep some kind of reference counter for that and then we can expire the go routine if it become hopeless?  

I'm not sure what the best answer would be, but something I just thought of (may or may not work) would be an annotation acting as a counter on the dv maybe?  Or  we do something like an annotation with a time-stamp (lastDetectedFailure?) where we see a volume has been in fail for more than some arbitrary amount of time we terminate the routine here?  If it goes to another state before expiration we clear the annotation

Reply:
We can keep variables outside of the loop to keep some state, so there is no need to put things in annotations to keep state. The tricky part would be, how do we define what is 'long' enough in either counter or time. But those are interesting ideas, let me mull them over.